src/library/scala/collection/mutable/HashSet.scala

```diff
@@ -203,6 +203,40 @@ final class HashSet[A](initialCapacity: Int, loadFactor: Double)
     }
   }
 
+  override def filterInPlace(p: A => Boolean): this.type = {
+    if (nonEmpty) {
+      var bucket = 0
+
+      while (bucket < table.length) {
+        var head = table(bucket)
```

Comment:
Are you sure this won't impact performance at all?

Reply:
To be clear, I was thinking of an inner method, nested within the `filterInPlace` or the `if` branch; since that would only have one call-site, and since it is not recursive, it should not be difficult for the compiler to inline it. 

However, to answer your question, I am not sure as to how well does the compiler optimise such inner function calls.